lib/ui/assets.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.10
+
+part of dart.ui;
+
+/// Load the asset bytes specified by [assetKey].
+///
+/// The [assetKey] is generally the filepath of the asset which is bundled
+/// into the flutter application. This API is not supported on the Web.
+///
+/// If the [assetKey] does not correspond to a real asset, returns `null`.
+ByteData? loadAsset(String assetKey) {
```

Comment:
If we use `ImmutableBuffer` here, we could have a nice fast path for loading these without relying on Dart.

But right now, that class is only useful when creating images. We'd want to consider extending it in some way so you could get a `ByteData` from it, or a Uint8List or something.

Reply:
Today, we ahve `ImmutableBuffer.fromUint8List`, but this could allow for creating `ImmutableBuffer`s directly, which would definitely be useful for e.g. `Image.asset`.

Reply:
Sorry I'm not remembering all of the context around `ImmutableBuffer`. If we had a `ReadonlyUint8List` from the core Dart SDK, would `ImmutableBuffer` still be needed?

Reply:
Ultimately we need a way to somewhat compatible with the existing API and expose the bytes to users applications. Assets are used for more than just images today, we have APIs to load strings/bytedata/json

Reply:
that's not to say we couldn't add a special API for images though, they are one of the most common cases

Reply:
Agreed.

ImmutableBuffer was added to avoid having to do multiple copies of image data from a Uint8List to an SkData buffer if it ends up being reused.

If the Dart SDK has an appropriate type that's useful outside of dart:ui, that's probably better. But if not we could expand ImmutableBuffer to have access as ByteData (which, when Dart added some type appropriate for it, could just become a pass through to that type).